### PR TITLE
Enhancement/50 response builder and rendering

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: src/main/java/com/exasol/adapter/response/CreateVirtualSchemaResponse.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package com.exasol.adapter.response;
+
+import com.exasol.adapter.metadata.SchemaMetadata;
+import com.exasol.adapter.response.converter.ResponseJsonConverter;
+
+/**
+ * Used in overloaded convert() method of {@link ResponseJsonConverter}
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/CreateVirtualSchemaResponse.java]: "This class is an abstract representation of a response created by a Virtual Schema Adapter as result of a request to create a new virtual schema."

File: src/main/java/com/exasol/adapter/response/DropVirtualSchemaResponse.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.exasol.adapter.response;
+
+import com.exasol.adapter.response.converter.ResponseJsonConverter;
+
+/**
+ * Used in overloaded convert() method of {@link ResponseJsonConverter}
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/DropVirtualSchemaResponse.java]: Change as mentioned above.

File: src/main/java/com/exasol/adapter/response/CreateVirtualSchemaResponse.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    public static class Builder {
+        SchemaMetadata schemaMetadata;
+
+        /**
+         * Add the Virtual Schema's metadata
+         *
+         * @return builder instance for fluent programming
+         */
+        public Builder addSchemaMetadata(final SchemaMetadata schemaMetadata) {
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/CreateVirtualSchemaResponse.java]:
```java
public Builder schemaMetadata(final SchemaMetadata schemaMetadata) {
```

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    /**
+     * Converts drop response into a Json format
+     *
+     * @param dropResponse instance
+     * @return string representation of a JsonObject
+     */
+    @SuppressWarnings("squid:S1172")
+    public String convert(final DropVirtualSchemaResponse dropResponse) {
+        final JsonBuilderFactory factory = JsonHelper.getBuilderFactory();
+        final JsonObject res = factory.createObjectBuilder().add("type", "dropVirtualSchema").build();
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: Please don't abbreviate.

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import com.exasol.adapter.response.CreateVirtualSchemaResponse;
+import com.exasol.adapter.response.DropVirtualSchemaResponse;
+import com.exasol.utils.JsonHelper;
+
+import javax.json.Json;
+import javax.json.JsonBuilderFactory;
+import javax.json.JsonObject;
+
+/**
+ * Converts response into Json format
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: "... JSON format ..."
Please correct in other JavaDoc too.

File: src/main/java/com/exasol/adapter/response/CreateVirtualSchemaResponse.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/CreateVirtualSchemaResponse.java]: fixed

File: src/main/java/com/exasol/adapter/response/DropVirtualSchemaResponse.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/DropVirtualSchemaResponse.java]: fixed

File: src/main/java/com/exasol/adapter/response/CreateVirtualSchemaResponse.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/CreateVirtualSchemaResponse.java]: fixed

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: fixed

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java (hunk 5 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: fixed

File: src/main/java/com/exasol/adapter/response/PushDownResponse.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+     * Get a {@link PushDownResponse} builder
+     *
+     * @return builder instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Get push down sql
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/PushDownResponse.java]: "SQL"

File: src/main/java/com/exasol/adapter/response/PushDownResponse.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+     *
+     * @return push down sql string
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/PushDownResponse.java]: "SQL"

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    /**
+     * Converts create virtual schema response into a JSON format
+     *
+     * @param createResponse instance
+     * @return string representation of a JSON Object
+     */
+    public String convertCreateVirtualSchemaResponse(final CreateVirtualSchemaResponse createResponse) {
+        final JsonObject response = Json.createObjectBuilder() //
+              .add("type", "createVirtualSchema") //
+              .add(SCHEMA_METADATA, SchemaMetadataSerializer.serialize(createResponse.getSchemaMetadata())) //
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: Please write a ticket that we are going to replace the static util class `SchemaMetadataSerializer` by a dedicated JSON builder.

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+    /**
+     * Converts drop virtual schema response into a JSON format
+     *
+     * @param dropResponse instance
+     * @return string representation of a JSON Object
+     */
+    @SuppressWarnings("squid:S1172")
+    public String convertDropVirtualSchemaResponse(final DropVirtualSchemaResponse dropResponse) {
+        final JsonBuilderFactory factory = JsonHelper.getBuilderFactory();
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: Check whether one factory is enough for all invocations in this class. In that case build it during construction.

File: src/main/java/com/exasol/adapter/response/PushDownResponse.java (hunk 6 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/PushDownResponse.java]: fixed

File: src/main/java/com/exasol/adapter/response/PushDownResponse.java (hunk 7 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/PushDownResponse.java]: fixed

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java (hunk 9 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: fixed

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java (hunk 8 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: done

File: src/main/java/com/exasol/adapter/response/AbstractResponse.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
+package com.exasol.adapter.response;
+
+import com.exasol.adapter.metadata.SchemaMetadata;
+
+/**
+ * This class is an abstract representation of a response
+ * created by a Virtual Schema Adapter.
+ */
+public class AbstractResponse {
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/AbstractResponse.java]: Remove `public`

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+     * Converts get capabilities response into a JSON format
+     *
+     * @param getCapabilitiesResponse instance
+     * @return string representation of a JSON Object
+     */
+    public String convertGetCapabilitiesResponse(final GetCapabilitiesResponse getCapabilitiesResponse) {
+        final JsonObjectBuilder builder = Json.createObjectBuilder().add("type", "getCapabilities");
+        final JsonArrayBuilder arrayBuilder = Json.createArrayBuilder();
+        final Capabilities capabilities = getCapabilitiesResponse.getCapabilities();
+        for (final MainCapability capability : capabilities.getMainCapabilities()) {
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: Extract this and the next four for loops into a separate method each.

File: src/main/java/com/exasol/adapter/response/AbstractResponse.java (hunk 10 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/AbstractResponse.java]: done

File: src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java (hunk 11 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java]: done

**Pull request merged**